core/unit_test/TestScan.hpp

```diff
@@ -170,11 +170,108 @@ struct TestScan {
   }
 };
 
+template <class Device, class T>
+struct TestSmallSizeTypeScan {
+  using execution_space = Device;
+  using value_type      = T;
+
+  TestSmallSizeTypeScan(const size_t N, bool use_total) : x("in", N) {
+    using exec_policy = Kokkos::RangePolicy<execution_space>;
+
+    Kokkos::View<int, Device> errors_a("Errors");
+    Kokkos::deep_copy(errors_a, 0);
+    errors = errors_a;
+
+    Kokkos::deep_copy(x, 1);
+
+    T res = T(0);
+
+    if (use_total)
+      Kokkos::parallel_scan(exec_policy(0, N), *this, res);
+    else
+      Kokkos::parallel_scan(exec_policy(0, N), *this);
+    Kokkos::fence();
+
+    check_error(res, N, use_total);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(int i, value_type& update, bool final) const {
+    const auto val = x(i);
+    if (final) {
+      x(i) = update;
+
+      // We should have a nice count from 0 to 1...
+      if (update != static_cast<value_type>(i)) {
+        int fail = errors()++;
+
+        // Limit the amount of output
+        if (fail < 20) {
+          KOKKOS_IMPL_DO_NOT_USE_PRINTF(
+              "TestSmallSizeTypeScan(%d) = %ld != %ld\n", i,
+              static_cast<long>(update), static_cast<long>(i));
+        }
+      }
```

Comment:
Do not mix computation and verification.  Launch another kernel for that.

Reply:
Similarly, should I remove from the existing test?